Sources/MarathonCore/Export.swift

```diff
@@ -0,0 +1,123 @@
+        }
+        
+        // Script Name or Full Script Path given?
+        let isScriptName = scriptNameOrPath.components(separatedBy: "/").count == 1
+        
+        let file: File
+        if isScriptName {
+            // Does File exist in current directory?
+            file = try FileSystem().currentFolder.file(atPath: scriptNameOrPath)
+        } else {
+            // Does File exist at full path?
+            file = try FileSystem().createFile(at: scriptNameOrPath)
+        }
+        
+        // Is the File a script managed by marathon?
+        let isManagedScript = !(scriptManager.managedScriptPaths.filter { $0 == file.path }.isEmpty)
+        guard isManagedScript else {
+            throw Error.noMarathonScriptFound
+        }
+        
+        let exportFolder: Folder
+        if arguments.count > 1 {
+            // If more than one argument, try to create Folder at export-path
+            exportFolder = try FileSystem().createFolderIfNeeded(at: arguments[1])
+        } else {
+            // Otherwise use default export path
+            exportFolder = FileSystem().currentFolder
+        }
+        
+        let exportPath = exportFolder.path.appending(file.name.asScriptName())
+        if exportFolder.containsSubfolder(named: file.name.asScriptName()) {
+            throw Error.failedToExportScript("\(file.name). Directory already exists at path: \(exportPath)") // TODO: How to handle this?
+        }
+        
+        try export(file, to: exportFolder)
+    }
+    
+    private func export(_ file: File, to folder: Folder) throws {
+        guard let sourcesDir = try? folder.createSubfolder(named: "Sources") else {
+            throw Error.failedToExportScript(file.name)
+        }
+        
+        guard let scriptFileData = try? file.read() else {
+            throw Error.failedToExportScript(file.name)
+        }
+        
+        // TODO: - add inline dependecies
+        try sourcesDir.createFile(named: file.name, contents: scriptFileData)
+        
+        guard let packageFile = try? folder.createFile(named: "Package.swift") else {
+            throw Error.failedToExportScript(file.name)
+        }
+        
+        let script = try scriptManager.script(at: file.path)
+        let packageDescription = try packageManager.makePackageDescription(for: script)
+        let packageComponents = packageDescription.components(separatedBy: .newlines)
+        let packageNameComponents = packageComponents[3].components(separatedBy: ":")
+        let newPackageNameComponents = [packageNameComponents[0],  " \"\(file.name.asScriptName())\""]
+        let newPackageName = newPackageNameComponents.joined(separator: ":")
+        var newPackageComponents = packageComponents
+        newPackageComponents.remove(at: 3)
+        newPackageComponents.insert(newPackageName, at: 3)
+        let newPackageDescription = newPackageComponents.joined(separator: "\n")
+        try packageFile.write(string: newPackageDescription)
```

Comment:
Yeah, I think what we want to do is this:

1. Check what dependencies **this script** is using (compared to all of Marathon's installed packages), by using the inline dependency resolver (kind of). Right now `resolveInlineDependencies(from:)` checks for the inline dependency annotations, but the code that loops over and identifies `imports` could be extracted to a new public API that could be used for this. I'm thinking something like `scriptManager.resolveImports(from:)`. And then, internally within `ScriptManager`, we could use that new API in `resolveInlineDependencies` as well.

2. Look up all dependencies in Marathon's package cache, to identify their major version + URL.

3. Generate a `Package.swift` based on the gathered information.